Library/Homebrew/utils/curl.rb

```diff
@@ -484,10 +504,24 @@ def parse_curl_response(response_text)
       response_text = response_text.sub(%r{^HTTP/.* (\d+).*$\s*}, "")
 
       # Create a hash from the header lines
-      response[:headers] =
-        response_text.split("\r\n")
-                     .to_h { |header| header.split(/:\s*/, 2) }
-                     .transform_keys(&:downcase)
+      response[:headers] = {}
+      response_text.split("\r\n").each do |line|
+        header_name, header_value = line.split(/:\s*/, 2)
+        header_name = header_name&.strip&.downcase
+        header_value&.strip!
+        next if header_name.blank?
```

Comment:
```suggestion
        header_name, header_value = line.split(/:\s*/, 2)
        next if header_name.blank?

        header_name = header_name.strip.downcase
        header_value&.strip!
```

and do you want to skip if `header_value` is `nil` or not?

Reply:
Good call. The original code was intended to work around a `header_name` string that only contains whitespace but I briefly forgot that `#blank?` handles this as well.

> do you want to skip if `header_value` is `nil` or not?

We want to skip when `header_value` is `nil` but not when it's an empty string. However, this situation will be handled by `next if header_name.blank?`, as `header_name` will always be `nil` when `header_value` is `nil`.

That is to say, these values are only `nil` when `line` doesn't match the regex, in which case they're both `nil`. If we encountered a header like `something:`, `header_value` would be an empty string, which is fine (i.e., feels correct to include a header with a blank value). [In the extreme case, a `line` like `:` would lead to both `header_name` and `header_value` being empty strings but `next if header_name.blank?` accounts for this.]

-----

I'm going to push this change but I'll be away for the next few hours, so I'll wait to merge this until I return. If it's good with you, I would like to add some additional tests to better exercise these particular lines, as I don't think I covered all the possible scenarios. It shouldn't be anything notable but would it be fine to merge with the additional tests? If not, I'll just leave them out in favor of merging.

Reply:
👍🏻 sounds good.



> It shouldn't be anything notable but would it be fine to merge with the additional tests?

Fine with me either way!

Reply:
> It shouldn't be anything notable

Famous last words 😆. In the process of adding the additional tests and checking coverage, I ended up tweaking bits of `#parse_curl_response`. The changes aren't very substantial but it's enough that I'm not comfortable adding them here without further review.

I'm going to go ahead and merge this as-is and either create a follow-up PR for the additional tests or simply include it as part of a follow-up PR for the refactoring ideas (i.e., a `Response` struct/class or type aliases). There's no rush to add these additional tests (they just cover malformed response text), so they'll keep until you return.